File: Sorting Algorithm/CPP/InsertionSort.cpp
#include <iostream> 
using namespace std; 
  
void insertionSort(int arr[], int n)  
{  
    int i, key, j;  
    for (i = 1; i < n; i++) 
    {  
        key = arr[i];  
        j = i - 1;  
  
        while (j >= 0 && arr[j] > key) 
        {  
            arr[j + 1] = arr[j];  
            j = j - 1;  
        }  
        arr[j + 1] = key;  
    }  
}  
  

int main(void)  
{  
    const int n = 7;
    int arr[] = {89, 45, 68, 90, 29, 34, 17};    
    insertionSort(arr, n); 
     
     for (int i = 0; i < n; i++)  
        cout << arr[i] << " ";   
  
    return 0;  
}  

File: Sorting Algorithm/JavaScript/InsertionSort.js
const insertionSort = (arr) => {
    for (let i = 1; i < arr.length; i++) {
      let j = i - 1
      let tmp = arr[i]
      while (j >= 0 && arr[j] > tmp) {
        arr[j + 1] = arr[j]
        j--
      }
      arr[j+1] = tmp
    }
    return arr
  }

let arr = [89,45,68,90,29,34,17]
let sortedArray = insertionSort(arr)
console.log(sortedArray)